File: problem-solving/solutions/temp/test/blind75/ValidAnagramTest.java
package blind75;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import xyz.grind.coding.blind75.ValidAnagram;

import static org.junit.jupiter.api.Assertions.assertAll;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class ValidAnagramTest {

    @Test
    @DisplayName("Should return true if two strings are anagrams")
    void shouldReturnTrueIfFoundDuplicates() {
        assertAll(
            () -> assertTrue(ValidAnagram.isAnagramSortingBruteForce("silent", "listen")),
            () -> assertTrue(ValidAnagram.isAnagramSortingBruteForce("anagram", "gaanrma")),
            () -> assertTrue(ValidAnagram.isAnagramHashSet("silent", "listen")),
            () -> assertTrue(ValidAnagram.isAnagramHashSet("anagram", "gaanrma")),
            () -> assertTrue(ValidAnagram.isAnagramHashSet("art", "tra")),
            () -> assertTrue(ValidAnagram.isAnagram("silent", "listen")),
            () -> assertTrue(ValidAnagram.isAnagram("anagram", "gaanrma")),
            () -> assertTrue(ValidAnagram.isAnagram("art", "tra")),
            () -> assertTrue(ValidAnagram.isAnagram("aps", "pas"))
        );
    }

    @Test
    @DisplayName("Should return false if it two strings aren't anagrams")
    void shouldReturnFalseIfNoDuplicates() {
        assertAll(
            () -> assertFalse(ValidAnagram.isAnagramSortingBruteForce("silent", "isten")),
            () -> assertFalse(ValidAnagram.isAnagramSortingBruteForce("anagram", "aanrma")),
            () -> assertFalse(ValidAnagram.isAnagramHashSet("silent", "isten")),
            () -> assertFalse(ValidAnagram.isAnagramHashSet("anagram", "aanrma")),
            () -> assertFalse(ValidAnagram.isAnagramHashSet("art", "ra")),
            () -> assertFalse(ValidAnagram.isAnagram("silent", "isten")),
            () -> assertFalse(ValidAnagram.isAnagram("anagram", "aanrma")),
            () -> assertFalse(ValidAnagram.isAnagram("art", "ra")),
            () -> assertFalse(ValidAnagram.isAnagram("aps", "as"))
        );
    }

}

File: problem-solving/solutions/leetcode/blind75/src/test/kotlin/easy/ValidAnagramTest.kt
package easy

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.function.Executable
import xyz.grind.coding.easy.ValidAnagram

class ValidAnagramTest {
    @Test
    fun `Should return true if two strings are anagrams`() {
        Assertions.assertAll(
            Executable { assertTrue(ValidAnagram.isAnagramSortingBruteForce("silent", "listen")) },
            Executable { assertTrue(ValidAnagram.isAnagramSortingBruteForce("anagram", "gaanrma")) },
            Executable { assertTrue(ValidAnagram.isAnagramHashSet("silent", "listen")) },
            Executable { assertTrue(ValidAnagram.isAnagramHashSet("anagram", "gaanrma")) },
            Executable { assertTrue(ValidAnagram.isAnagramHashSet("art", "tra")) },
            Executable { assertTrue(ValidAnagram.isAnagram("silent", "listen")) },
            Executable { assertTrue(ValidAnagram.isAnagram("anagram", "gaanrma")) },
            Executable { assertTrue(ValidAnagram.isAnagram("art", "tra")) },
            Executable { assertTrue(ValidAnagram.isAnagram("aps", "pas")) },
            Executable { assertTrue(ValidAnagram.isAnagramAdjusted("silent", "listen")) },
            Executable { assertTrue(ValidAnagram.isAnagramAdjusted("anagram", "gaanrma")) },
            Executable { assertTrue(ValidAnagram.isAnagramAdjusted("art", "tra")) },
            Executable { assertTrue(ValidAnagram.isAnagramAdjusted("aps", "pas")) },
        )
    }

    @Test
    fun `Should return false if it two strings aren't anagrams`() {
        Assertions.assertAll(
            Executable { assertFalse(ValidAnagram.isAnagramSortingBruteForce("silent", "isten")) },
            Executable { assertFalse(ValidAnagram.isAnagramSortingBruteForce("anagram", "aanrma")) },
            Executable { assertFalse(ValidAnagram.isAnagramHashSet("silent", "isten")) },
            Executable { assertFalse(ValidAnagram.isAnagramHashSet("anagram", "aanrma")) },
            Executable { assertFalse(ValidAnagram.isAnagramHashSet("art", "ra")) },
            Executable { assertFalse(ValidAnagram.isAnagram("silent", "isten")) },
            Executable { assertFalse(ValidAnagram.isAnagram("anagram", "aanrma")) },
            Executable { assertFalse(ValidAnagram.isAnagram("art", "ra")) },
            Executable { assertFalse(ValidAnagram.isAnagram("aps", "as")) },
            Executable { assertFalse(ValidAnagram.isAnagramAdjusted("silent", "isten")) },
            Executable { assertFalse(ValidAnagram.isAnagramAdjusted("anagram", "aanrma")) },
            Executable { assertFalse(ValidAnagram.isAnagramAdjusted("art", "ra")) },
            Executable { assertFalse(ValidAnagram.isAnagramAdjusted("aps", "as")) },
        )
    }
}
